### ISSUE TITLE
[BLOCKER] Implement Real-Time Training Monitoring with Progress Visualization

### ISSUE THREAD

**Issue by frankbria** [frankbria/narrative-modeling-app#76]
## 🎯 Overview
Implement real-time monitoring and visualization for model training progress with WebSocket-based updates.

## 📊 Current State
- No visibility into training progress
- Users don't know if training is working or stuck
- No way to track multiple parallel training jobs

## 🎯 Expected Outcome
Users should see:
- Real-time progress for each model being trained
- Current training metrics (accuracy, loss, etc.)
- Estimated time remaining
- Resource usage (CPU, memory)
- Training logs and error messages
- Ability to cancel long-running training

## ✅ Acceptance Criteria
- [ ] Progress dashboard showing all training models
- [ ] For each model:
  - Progress bar (0-100%)
  - Current stage (data prep, training, validation, etc.)
  - Current metrics (updating in real-time)
  - Time elapsed and estimated remaining
  - Resource usage indicators
- [ ] Training logs panel with:
  - Real-time log streaming
  - Severity levels (info, warning, error)
  - Searchable/filterable logs
- [ ] Alerts for:
  - Training completion
  - Errors or failures
  - Performance warnings (overfitting detected, etc.)
- [ ] Controls:
  - Cancel training button
  - Pause/resume (if supported)
  - View detailed logs
- [ ] Historical training runs browser

### Frontend (TypeScript/Next.js)
- Components:
  - \`TrainingMonitor\`: Main monitoring dashboard
  - \`ModelTrainingCard\`: Individual model progress
  - \`ProgressIndicator\`: Visual progress bar with stages
  - \`MetricsChart\`: Real-time metrics line chart
  - \`ResourceUsage\`: CPU/memory usage visualization
  - \`TrainingLogs\`: Log viewer with filtering
- WebSocket client for real-time updates
- State management for multiple concurrent training jobs

### Backend (Python/FastAPI)
- WebSocket endpoint: \`/ws/training/{training_id}\`
- Service:
  ```python
  class TrainingMonitor:
      async def emit_progress(
          self,
          training_id: str,
          progress: TrainingProgress
      ) -> None:
          \"\"\"Emit progress update via WebSocket\"\"\"
          pass
      
      async def emit_log(
          self,
          training_id: str,
          log: LogEntry
      ) -> None:
          \"\"\"Emit log entry via WebSocket\"\"\"
          pass
      
      async def get_training_status(
          self,
          training_id: str
      ) -> TrainingStatus:
          \"\"\"Get current training status\"\"\"
          pass
  
  class TrainingProgress(BaseModel):
      training_id: str
      model_name: str
      stage: str  # 'preprocessing' | 'training' | 'validation' | 'complete'
      progress_percent: float
      current_metrics: Dict[str, float]
      time_elapsed: float
      time_remaining_estimate: float
      resource_usage: ResourceUsage
  
  class LogEntry(BaseModel):
      timestamp: datetime
      level: str  # 'info' | 'warning' | 'error'
      message: str
      context: Dict[str, Any]
  
  class ResourceUsage(BaseModel):
      cpu_percent: float
      memory_mb: float
      gpu_percent: Optional[float]
  ```
- Background task integration with Celery
- Progress tracking in training loops
- Resource monitoring with psutil

### Monitoring Implementation
- Emit progress updates every N iterations
- Track training metrics per epoch/iteration
- Monitor system resources
- Store all logs in database for history
- Implement connection pooling for WebSocket

## 📋 Subtasks
- [ ] Implement WebSocket server endpoint
- [ ] Create progress tracking in training loops
- [ ] Add resource monitoring
- [ ] Build logging system with structured logs
- [ ] Create frontend WebSocket client
- [ ] Build training monitor dashboard
- [ ] Implement metrics charting
- [ ] Add cancel training functionality
- [ ] Create historical training runs viewer
- [ ] Add unit tests for monitoring service
- [ ] Create integration tests for WebSocket communication
- [ ] Add E2E tests for monitoring UI
- [ ] Update documentation

## 🔗 Dependencies
- #75 (Core AutoML engine) - provides training process to monitor

## 📚 Related Issues
- Core AutoML engine (#75)
- Model versioning (to be created)

## 🏷️ Labels
\`blocker\`, \`backend\`, \`frontend\`, \`stage-5-model-training\`, \`realtime\`

## ⏱️ Estimated Effort
2-3 weeks

## 📝 Notes
- Essential for user confidence during training
- WebSocket performance critical for responsiveness
- Consider fallback to polling if WebSocket fails
- Add telemetry for training patterns
- Follow TDD approach with >85% test coverage
